packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-actions.tsx

```diff
@@ -109,11 +112,23 @@ export const PipelineActions: React.FunctionComponent<PipelineActionsProps> = ({
   );
 };
 
-const mapState = ({ pipeline }: RootState) => ({
-  isPipelineInvalid: pipeline.some(
-    (x) => x.isEnabled && (!x.isValid || x.error)
-  ),
-});
+const mapState = ({ pipeline }: RootState) => {
+  const resultPipeline = pipeline.filter(
+    (stageState) => !isEmptyishStage(stageState)
+  );
+  const lastStage = resultPipeline[resultPipeline.length - 1];
+  const isMergeOrOutPipeline = ['$merge', '$out'].includes(
+    lastStage?.stageOperator
+  );
+  const isPipelineInvalid = resultPipeline.some(
+    (stageState) => !stageState.isValid || Boolean(stageState.error)
+  );
+
+  return {
+    isRunButtonDisabled: isPipelineInvalid,
```

Comment:
Should the run button be disabled if the pipeline is empty?

Reply:
Ah, yeah, I guess we can disable it, good catch! Technically it is a valid aggregation though 😄